File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int maxdif=0;
    int min=prices[0];
    for(int i=1; i<pricesSize; i++){
        int cur=prices[i];
        if(min>cur){
            min=cur;
        }
        int curdif=cur-min;
        if(curdif>maxdif){
            maxdif=curdif;
        }
    }
    return maxdif;
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxprof=0
        curlow=prices[0]
        for x in prices:
            curlow=min(curlow,x)
            diff=x-curlow
            maxprof=max(diff,maxprof)
        return maxprof    